core/src/main/java/org/mule/config/ImmutableThreadingProfile.java

```diff
@@ -162,7 +162,7 @@ public void setWorkManagerFactory(WorkManagerFactory workManagerFactory)
 
     public WorkManager createWorkManager(String name, int shutdownTimeout)
     {
-        return workManagerFactory.createWorkManager(new ImmutableThreadingProfile(this), name, shutdownTimeout);
+        return workManagerFactory.createWorkManager(this, name, shutdownTimeout);
```

Comment:
The clone still happens, just further down instead of being repeated in every threadProfile implementation.